### DISCUSSION TITLE
Will start to refactor socketstuff

### DISCUSSION THREAD

**Discussion by togfoxy** [togfoxy/MarsLander#164] [General]
I think the main branch is as good as it can be for the moment.

My personal priorities are:
- to respond to PR's as best I can and keep the main branch as current as possible
- refactor socketstuff to align to style guide
- fix the slow downs relating to terrain/base generation
- get multiplayer to work over LAN

I'm dabbling with one or more computer opponents to make the game a bit more interesting for single players.

Will need to think about game modes at some point. I'm currently working on a 'default' game mode of "race to the end" knowing there is no end.
@Philbywhizz @MadByteDE

**Comment by MadByteDE** [comment]
Sounds good. 

Here's what I'd like to do next:

- Implementing new lander physics model (#148)
- Implementing asset loader
- Find better ways for organizing code to get rid of globals

I'm still thinking about working on tile-based terrain generation in the background as well. But I don't think that this is something that can be ready for at least a couple weeks and isn't much of an priority rn.

Also, I'd like to concentrate more on graphical improvements but they would need to be different from [the concept I made previously ](https://i.imgur.com/s42bV82.png) because otherwise we'd need to work more on resolution, scaling and change a lot of different stuff like object sizes, world generation etc. Maybe that's something best suited whenever the code is in a more versatile state.

**Comment by Philbywhizz** [comment]
Looks like you guys have a plan in place already. Go for it.

As for myself, I haven't really decided what I want to focus on at this stage. At the moment I've just been looking at the existing code base and if I see something that is broken or needs improvement (not just game play, it could be menu layouts, look and feel, etc).

I tend to focus on the 'other areas' that most devs don't like (or have time) doing (documentation, wikis, project management, quality, organization, etc).

**Reply by togfoxy** [reply]
Seems like we have something for everyone. :)

**Comment by togfoxy** [comment]
I have rewritten the socket code and I'm playing mplayer on a LAN with buddy.

It is great!

* giggle *

**Reply by MadByteDE** [reply]
Awesome, good job. :grin:
